intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp

```diff
@@ -0,0 +1,322 @@
+  return id;
+}
+
+int Model::quant_model(const std::string& model_path, const std::string& out_path, int bits, const std::string& alg,
+                       int block_size, const std::string& scale_dtype, const std::string& compute_type) {
+  quant_params q_params;
+#ifdef MODEL_NAME
+  q_params.model_name = MODEL_NAME;
+#endif
+  model_archs mt = model_name_to_arch::init().find(q_params.model_name);
+  if (mt == MODEL_UNKNOWN) {
+    fprintf(stderr, "error, please set model_name \n");
+    exit(0);
+  }
+  q_params.model_arch = mt;
+  q_params.model_file = model_path;
+  q_params.out_file = out_path;
+  q_params.bits = bits;
+  q_params.alg = alg;
+  q_params.block_size = block_size;
+  q_params.scale_dtype = scale_dtype;
+  q_params.compute_type = compute_type;
+
+  ne_ftype ftype = quant_params_to_ftype(q_params);
+  printf("ne_ftype: %d\n", ftype);
+  const int nthread = q_params.nthread;
+
+  auto quant_layer = get_model_quant_layer(q_params.model_name);
+  if (model_quantize(q_params, quant_layer)) {
+    fprintf(stderr, "%s: failed to quantize model from '%s'\n", __func__, q_params.model_file.c_str());
+    return 1;
+  }
+  return 0;
+}
+
+namespace py = pybind11;
+
+#if MODEL_NAME_ID == 1
+
+PYBIND11_MODULE(gptj_cpp, m)
+#elif MODEL_NAME_ID == 2
+
+PYBIND11_MODULE(falcon_cpp, m)
+
+#elif MODEL_NAME_ID == 3
+
+PYBIND11_MODULE(gptneox_cpp, m)
+
+#elif MODEL_NAME_ID == 4
+
+PYBIND11_MODULE(dolly_cpp, m)
+
+#elif MODEL_NAME_ID == 5
+
+PYBIND11_MODULE(llama_cpp, m)
+
+#elif MODEL_NAME_ID == 6
+
+PYBIND11_MODULE(mpt_cpp, m)
+
+#elif MODEL_NAME_ID == 7
+
+PYBIND11_MODULE(starcoder_cpp, m)
+
+#elif MODEL_NAME_ID == 8
+
+PYBIND11_MODULE(opt_cpp, m)
+
+#elif MODEL_NAME_ID == 9
+
+PYBIND11_MODULE(bloom_cpp, m)
+
+#elif MODEL_NAME_ID == 10
+
+PYBIND11_MODULE(chatglm2_cpp, m)
+
+#elif MODEL_NAME_ID == 11
+
+PYBIND11_MODULE(chatglm_cpp, m)
+
+#endif
+{
+  m.doc() = "cpp model python binding";
+  py::class_<Model>(m, "Model", py::module_local())
+      .def(py::init())
+      .def("init_model", &Model::init_model, "initial model with model path and parameters",
+                          py::arg("model_path"),
+                          py::arg("max_new_tokens") = -1,
+                          py::arg("batch_size") = 512,
+                          py::arg("ctx_size") = 512,
+                          py::arg("seed") = -1,
+                          py::arg("threads") = 8,
+                          py::arg("repeat_penalty") = 1.1f,
+                          py::arg("post_process") = "topk"
+                          )
+      .def("generate", &Model::generate, "Generate tokens with prompt",
+                          py::arg("prompt"),
+                          py::arg("sentence_mode") = true
+                          )
+      .def_static("quant_model", &Model::quant_model, "Quantize model",
+                          py::arg("model_path"),
+                          py::arg("out_path"),
+                          py::arg("bits") = 4,
```

Comment:
so arg `bits` == arg `weight_dtype`? If so, why not change `bits` to `weight_dtype` for better understanding?

Reply:
we will change this on pr https://github.com/intel/intel-extension-for-transformers/pull/320